src/main/java/appeng/tile/storage/TileIOPort.java

```diff
@@ -277,87 +276,84 @@ public TickRateModulation tickingRequest( final IGridNode node, final int ticksS
-					AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
-			final IMEInventory<IAEFluidStack> fluidNet = this.getProxy().getStorage().getInventory(
-					AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
 			final IEnergySource energy = this.getProxy().getEnergy();
-			for( int x = 0; x < 6; x++ )
+			for( int x = 0; x < NUMBER_OF_CELL_SLOTS; x++ )
 			{
 				final ItemStack is = this.inputCells.getStackInSlot( x );
 				if( !is.isEmpty() )
 				{
-					if( ItemsToMove > 0 )
-					{
-						final IMEInventory<IAEItemStack> itemInv = this.getInv( is, AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
-						final IMEInventory<IAEFluidStack> fluidInv = this.getInv( is,
-								AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+					boolean shouldMove = true;
 
-						if( this.manager.getSetting( Settings.OPERATION_MODE ) == OperationMode.EMPTY )
+					for( IStorageChannel<? extends IAEStack<?>> c : AEApi.instance().storage().storageChannels() )
+					{
+						if( itemsToMove > 0 )
 						{
-							if( itemInv != null )
+							final IMEMonitor<? extends IAEStack<?>> network = this.getProxy().getStorage().getInventory( c );
+							final IMEInventory<?> inv = this.getInv( is, c );
+
+							if( inv == null )
 							{
-								ItemsToMove = this.transferContents( energy, itemInv, itemNet, ItemsToMove,
-										AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
+								continue;
 							}
-							if( fluidInv != null )
+
+							if( this.manager.getSetting( Settings.OPERATION_MODE ) == OperationMode.EMPTY )
 							{
-								ItemsToMove = this.transferContents( energy, fluidInv, fluidNet, ItemsToMove,
-										AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+								itemsToMove = this.transferContents( energy, inv, network, itemsToMove, c );
 							}
-						}
-						else
-						{
-							if( itemInv != null )
+							else
 							{
-								ItemsToMove = this.transferContents( energy, itemNet, itemInv, ItemsToMove,
-										AEApi.instance().storage().getStorageChannel( IItemStorageChannel.class ) );
+								itemsToMove = this.transferContents( energy, network, inv, itemsToMove, c );
 							}
-							if( fluidInv != null )
+
+							shouldMove &= this.shouldMove( inv );
+
+							if( itemsToMove > 0 )
 							{
-								ItemsToMove = this.transferContents( energy, fluidNet, fluidInv, ItemsToMove,
-										AEApi.instance().storage().getStorageChannel( IFluidStorageChannel.class ) );
+								ret = TickRateModulation.IDLE;
+							}
+							else
+							{
+								ret = TickRateModulation.URGENT;
 							}
 						}
+					}
 
-						if( ItemsToMove > 0 && this.shouldMove( itemInv, fluidInv ) && !this.moveSlot( x ) )
-						{
-							return TickRateModulation.IDLE;
-						}
-
-						return TickRateModulation.URGENT;
+					if( itemsToMove > 0 && shouldMove && this.moveSlot( x ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this conditional structure or edit its code blocks so that they're not all the same. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3923)